pkg/streamingpromql/planning.go

```diff
@@ -0,0 +1,663 @@
+// SPDX-License-Identifier: AGPL-3.0-only
```

Comment:
(tests for this file would be good, but fine for followup)

Reply:
Could you elaborate? Most of the functionality here should be covered by the tests in `planning_test.go` and the existing tests for the engine.

Reply:
sorry that comment should have been deleted. It was an oversight in reviewing the files in order.